contracts/DCAHubSwapper/DCAHubSwapperSwapHandler.sol

```diff
@@ -144,10 +170,50 @@ abstract contract DCAHubSwapperSwapHandler is DeadlineValidation, DCAHubSwapperP
 
   /// @inheritdoc IDCAHubSwapperSwapHandler
   function defineDexSupport(address _dex, bool _support) external onlyGovernor {
-    if (_dex == address(0)) revert IDCAHubSwapper.ZeroAddress();
+    if (_dex == address(0)) revert ZeroAddress();
     isDexSupported[_dex] = _support;
   }
 
+  function _handleSwapWithDexesCallback(IDCAHub.TokenInSwap[] calldata _tokens, bytes memory _data) internal {
+    SwapWithDexCallbackData memory _callbackData = abi.decode(_data, (SwapWithDexCallbackData));
+
+    // Validate that all swappers are allowlisted
+    for (uint256 i; i < _callbackData.swappers.length; i++) {
+      _assertSwapperIsAllowlisted(_callbackData.swappers[i]);
+    }
+
+    // Execute swaps
+    for (uint256 i; i < _callbackData.executions.length; i++) {
+      SwapExecution memory _execution = _callbackData.executions[i];
+      _callbackData.swappers[_execution.swapperIndex].functionCall(_execution.swapData, 'Call to swapper failed');
```

Comment:
Why not `_executeSwap`? Also, shouldn't we have an allowance check here (`_maxApproveSpenderIfNeeded`) ?

I think I'm a bit confused if this is going to be "hosting" a partial "swap proxy", or if here we are going to call `swap proxy`. Not sure.

Reply:
Because execute swap takes calldata bytes, and here they are in memory

Also, the allowance check will be done before calling the hub, so it's not needed ok the callback. By doing this, we can save transferring a lot of bytes to the hub and back

Reply:
Hey @0xged , I was just reviewing PRs and the approval is in #110

Reply:
@0xsambugs ohhhh. Ok. Looking at #110, although it took me a quite some time I made the click.